src/Components/Providers/ModalProvider.jsx

```diff
@@ -0,0 +1,30 @@
+import React, { createContext, useContext, useState } from 'react';
+
+const modalContext = createContext(null);
+
+const ModalProvider = ({ children }) => {
+    const [modalOpen, setModalOpen] = useState({
+        status: false,
+        data: {
```

Comment:
there are multiple values you are setting over here. I would suggest use reducer. it will be better